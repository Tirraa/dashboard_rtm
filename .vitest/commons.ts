export const INVALID_PATH = './bless/the/draco/these/rounds/holy/sorry/bout/your/luck';
