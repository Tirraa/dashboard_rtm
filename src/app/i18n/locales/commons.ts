export namespace CommonsVocab {
  export const brand = 'Rust Team Management';
  export const logo = 'logo'; // {ToDo} i18n this!
}

export default CommonsVocab;
