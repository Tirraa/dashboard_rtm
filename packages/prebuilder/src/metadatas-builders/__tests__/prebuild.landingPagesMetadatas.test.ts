// eslint-disable-next-line import/no-extraneous-dependencies
import { describe, expect, it } from 'vitest';
// eslint-disable-next-line import/no-extraneous-dependencies
import { INVALID_PATH } from '𝕍/commons';

import getLpMetadatas from '../landingPagesMetadatas';

const VALID_LP_FOLDER_PATH = './packages/prebuilder/src/metadatas-builders/__tests__/fake_lp_folder';

describe('getLpMetadatas', () => {
  it('should throw ENOENT, given invalid path', async () => {
    expect.assertions(1);

    try {
      await getLpMetadatas(INVALID_PATH);
    } catch (e) {
      const interceptedError = e as Error;
      if ('code' in interceptedError) {
        expect(interceptedError.code).toBe('ENOENT');
      } else {
        throw new Error('Error code not found');
      }
    }
  });

  it('should match snapshot', async () => {
    expect(await getLpMetadatas(VALID_LP_FOLDER_PATH)).toMatchSnapshot();
  });
});
