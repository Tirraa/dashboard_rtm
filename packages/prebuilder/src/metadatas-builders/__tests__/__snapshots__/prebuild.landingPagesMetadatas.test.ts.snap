// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getLpMetadatas > should match snapshot 1`] = `
{
  "fake-category": {
    "DEFAULT_LANGUAGE": [
      "fake-category-lp-00",
    ],
    "en": [
      "fake-category-lp-00",
    ],
  },
}
`;
